src/annoylib.h

```diff
@@ -554,36 +416,8 @@ struct Minkowski {
 struct Euclidean : Minkowski{
   template<typename T>
   static inline T distance(const T* x, const T* y, int f) {
-    T d = 0.0;
-    for (int i = 0; i < f; i++, x++, y++)
-      d += ((*x) - (*y)) * ((*x) - (*y));
-    return d;
-  }
-#ifdef USE_AVX
-  static inline float distance(const float* x, const float* y, int f) {
-    float result = 0;
-    int i = f;
-    if (f > 7) {
-      __m256 euclidean = _mm256_setzero_ps();
-      for (; i > 7; i -= 8) {
-        const __m256 x_minus_y = _mm256_sub_ps(_mm256_loadu_ps(x), _mm256_loadu_ps(y));
-        euclidean = _mm256_add_ps(euclidean, _mm256_mul_ps(x_minus_y, x_minus_y));
-        x += 8;
-        y += 8;
-      }
-      // Sum all floats in euclidean register.
-      result = hsum256_ps_avx(euclidean);
-    }
-    // Don't forget the remaining values.
-    for (; i > 0; i--) {
-      const float t = *x - *y;
-      result += t * t;
-      x++;
-      y++;
-    }
-    return result;
+    return dot(x, x, f) + dot(y, y, f) - 2 * dot(x, y, f);
```

Comment:
also possibly not quite as fast because of cache locality? feels like @a1k0n would know

Reply:
Yeah, if the various components of `x` and `y` are already loaded in registers, seems like you might want to roll them up while you have them.

Run a benchmark and see!

Reply:
already have a fix coming up, 1 sec